compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Context.scala

```diff
@@ -17,26 +19,26 @@ import com.daml.lf.speedy.SExpr
 import com.daml.lf.speedy.SValue
 import com.daml.lf.types.Ledger.Ledger
 import com.daml.lf.speedy.SExpr.{LfDefRef, SDefinitionRef}
-import com.daml.lf.validation.Validation
 import com.google.protobuf.ByteString
 
+import scala.collection.immutable.HashMap
+
 /**
   * Scenario interpretation context: maintains a set of modules and external packages, with which
   * scenarios can be interpreted.
   */
 object Context {
   type ContextId = Long
-  case class ContextException(err: String) extends RuntimeException(err, null, true, false)
+  case class ContextException(err: String) extends RuntimeException(err)
 
-  var nextContextId: ContextId = 0
-
-  def newContext(): Context = {
-    this.synchronized {
-      nextContextId += 1
-      new Context(nextContextId)
-    }
+  private val nextContextId: () => ContextId = {
```

Comment:
Maybe this is my lack of scala experience but I would find it easier to read if the `atmLong` was declared at the scope of the `Object`.

Reply:
fixed